src/platform/storage/posix/storage.c

```diff
@@ -262,7 +262,13 @@ mender_storage_get_artifact_name(const char **artifact_name) {
     mender_err_t ret = mender_storage_read_file(MENDER_STORAGE_NVS_ARTICACT_NAME, (void **)artifact_name, &artifact_name_length);
     if (MENDER_OK != ret) {
         if (MENDER_NOT_FOUND == ret) {
-            *artifact_name = "unknown";
+
+            /* Get the Artifact Name from the build, if set */
+            if (0 < strlen(CONFIG_MENDER_ARTIFACT_NAME)) {
```

Comment:
I find this comparison a bit confusing, TBH. The code formatting check doesn't require a comparison to have the constant on the left side so if you prefer `strlen(...) > 0` like me, please change it. If you like the above more, feel free to leave it like that.

Reply:
No, I don't like it neither! I just thought that we enforced these too :see_no_evil: